### PR TITLE
feat: add Member.has_any_role

### PR DESCRIPTION
**Pull request by luliola**
## Pull Request Type
- [x] Feature addition
- [x] Bugfix
- [ ] Documentation update
- [ ] Code refactor
- [ ] Tests improvement
- [ ] CI/CD pipeline enhancement
- [ ] Other: [Replace with a description]

## Description
This PR fixes has_any_role() not working on member object by adding it to user.py and makes it work as originally [documented](https://interactions-py.github.io/interactions.py/API%20Reference/API%20Reference/models/Internal/checks/#interactions.models.internal.checks.has_any_role).

## Changes
- Add has_any_role() to user.py

## Test Scenarios
Use has_any_role() on member object.

## Python Compatibility
- [x] I've ensured my code works on Python `3.10.x`
- [x] I've ensured my code works on Python `3.11.x`

## Checklist
- [x] I've run the `pre-commit` code linter over all edited files
- [x] I've tested my changes on supported Python versions
- [x] I've added tests for my code, if applicable
- [x] I've updated / added  documentation, where applicable

### CODE REVIEW

**Comment by AstreaTSS** [comment]
While I do appreciate the PR (because I think it's still useful), this isn't what the documentation says. It specifically says that it's for *checks*, not for the actual user.py, and never says that `Member` was supposed to have it.

Thus, this is fully a feature addition, not a bug fix.

**Review by AstreaTSS — CHANGES_REQUESTED** [review]
You'll need to add the typing of the function to `user.pyi` too, found in the same folder.

**Review by silasary — APPROVED** [review]

File: interactions/models/discord/user.py [hunk 1]
```diff
@@ -592,6 +592,15 @@ def has_role(self, *roles: Union[Snowflake_Type, Role]) -> bool:
...
         return all(to_snowflake(role) in self._role_ids for role in roles)
 
+    def has_any_role(self, roles: List[Union[Snowflake_Type, Role]]) -> bool:
+        """
+        Checks if the user has any of the given roles.
+
+        Args:
+            *roles: The Role(s) or role id(s) to check for
+        """
+        return any((self.has_role(to_snowflake(role)) for role in roles))
```
Comment (eightween) [inline interactions/models/discord/user.py:603]: I haven't tested this PR, but the top-down control flow for `to_snowflake()` doesn't make sense here for `roles` - `Role` ultimately inherits a `SnowflakeObject` but doesn't alias to `Snowflake_Type`. Has this code been tested?

Comment (AstreaTSS) [reply interactions/models/discord/user.py:603]: `Snowflake_Type` includes `SnowflakeObject` in its union.

**Review by eightween — APPROVED** [review]

**Comment by eightween** [comment]
PR is on track for becoming stale

**Review by AstreaTSS — APPROVED** [review]

**Pull request merged**